docs/getting_started_jax.ipynb

```diff
@@ -0,0 +1,669 @@
+        "id": "kLR4IUmXGb7b"
+      },
+      "source": [
+        "What we will want to do here sharding wise is:\n",
+        "\n",
+        "1. `data` parallelism on x\n",
+        "2. `model` parallelism on `w1` and `w2` through the [megatron](http://go/arxiv/1909.08053) sharding strategy.\n",
+        "\n",
+        "Now let's prepare the model for GSPMD sharding. Note that we will explicitly shard `w1`, but let GSPMD propagation shard `w2`."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 5,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "nZaM4mVKF1UQ",
+        "outputId": "929cb459-75ab-4030-9ca4-23f7f778bc2f"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "module @jit_predict attributes {mhlo.num_partitions = 8 : i32, mhlo.num_replicas = 1 : i32} {\n",
+            "  func.func public @main(%arg0: tensor<16x128xf32> {mhlo.sharding = \"{devices=[4,1,2]<=[8] last_tile_dim_replicate}\"}, %arg1: tensor<128x256xf32> {mhlo.sharding = \"{devices=[1,2,4]<=[4,2]T(1,0) last_tile_dim_replicate}\"}, %arg2: tensor<256x10xf32>) -> (tensor<16x10xf32> {jax.result_info = \"\"}) {\n",
+            "    %0 = stablehlo.tanh %arg0 : tensor<16x128xf32>\n",
+            "    %1 = stablehlo.dot_general %0, %arg1, contracting_dims = [1] x [0], precision = [DEFAULT, DEFAULT] : (tensor<16x128xf32>, tensor<128x256xf32>) -> tensor<16x256xf32>\n",
+            "    %2 = stablehlo.dot_general %1, %arg2, contracting_dims = [1] x [0], precision = [DEFAULT, DEFAULT] : (tensor<16x256xf32>, tensor<256x10xf32>) -> tensor<16x10xf32>\n",
+            "    %3 = stablehlo.sine %2 : tensor<16x10xf32>\n",
+            "    return %3 : tensor<16x10xf32>\n",
+            "  }\n",
+            "}\n",
+            "\n"
+          ]
+        }
+      ],
+      "source": [
+        "def run_in_out_shardings():\n",
+        "  samples = jax.ShapeDtypeStruct((16, 128), jnp.float32, sharding=NamedSharding(mesh, PartitionSpec('data', None)))\n",
+        "  samples_sharding = NamedSharding(mesh, PartitionSpec('data', None))\n",
+        "  w1 = jax.ShapeDtypeStruct((128, 256), jnp.float32, sharding=NamedSharding(mesh, PartitionSpec(None, 'model')))\n",
+        "  w1_sharding = NamedSharding(mesh, PartitionSpec(None, 'model'))\n",
+        "  w2 = jax.ShapeDtypeStruct((256, 10), jnp.float32)\n",
+        "  w2_sharding = UNSPECIFIED\n",
+        "\n",
+        "  print(jax.jit(predict, in_shardings=(samples_sharding, w1_sharding, w2_sharding)).lower(samples, w1, w2).as_text())\n",
+        "\n",
+        "run_in_out_shardings()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "B5oXRrVoPh8P"
+      },
+      "source": [
+        "GSPMD's sharding annotations are defined as [HloShardingV2](TBA). So we have the following correspondance:\n",
```

Comment:
HloShardingV2 has no public documentation, but is available publicly. I'd probably just say "GSPMD's sharding annotations look like the following:"